src/reader/odbc_reader.rs

```diff
@@ -463,15 +463,16 @@ impl OdbcReaderBuilder {
 
     /// No matter if the user explicitly specified a limit in row size, a memory limit, both or
     /// neither. In order to construct a reader we need to decide on the buffer size in rows.
+    /// If a schema is empty, return something non-zero to be allocated.
     fn buffer_size_in_rows(&self, bytes_per_row: usize) -> Result<usize, Error> {
-        let rows_per_batch = self.max_bytes_per_batch / bytes_per_row;
```

Comment:
Rather than a checked division, I would prefer to check whether `bytes_per_row == 0` beforehand.